backend/app/adapter/gqlapi/resolver/authmutation.go

```diff
@@ -90,19 +91,22 @@ func (a AuthMutation) CreateShortLink(args *CreateShortLinkArgs) (*ShortLink, er
 		return &ShortLink{shortLink: newShortLink}, nil
 	}
 
-	// TODO(issue#823): refactor error type checking
-	switch err.(type) {
-	case shortlink.ErrAliasExist:
+	var (
+		ae shortlink.ErrAliasExist
+		l shortlink.ErrInvalidLongLink
+		c shortlink.ErrInvalidCustomAlias
+		m shortlink.ErrMaliciousLongLink
+	)
+	if errors.As(err, &ae) {
 		return nil, ErrAliasExist(*customAlias)
-	case shortlink.ErrInvalidLongLink:
-		return nil, ErrInvalidLongLink{u.LongLink, string(err.(shortlink.ErrInvalidLongLink).Violation)}
-	case shortlink.ErrInvalidCustomAlias:
-		return nil, ErrInvalidCustomAlias{*customAlias, string(err.(shortlink.ErrInvalidCustomAlias).Violation)}
-	case shortlink.ErrMaliciousLongLink:
+	} else if errors.As(err, &l) {
+		return nil, ErrInvalidLongLink{u.LongLink, string(l.Violation)}
+	} else if errors.As(err, &c) {
+		return nil, ErrInvalidCustomAlias{*customAlias, string(c.Violation)}
+	} else if errors.As(err, &m) {
```

Comment:
```suggestion
	}
	if errors.As(err, &l) {
		return nil, ErrInvalidLongLink{u.LongLink, string(l.Violation)}
	}
	if errors.As(err, &c) {
		return nil, ErrInvalidCustomAlias{*customAlias, string(c.Violation)}
	}
	if errors.As(err, &m) {
```

Reply:
It seems that the else statements are unnecessary and would interferer readers from understanding the code.